Clean up SPDY.m

erikchen opened the pull request:
Fix a couple of retain/release bugs.

Begin to migrate the code to a more modern style that uses properties/synthesize rather than explicitly creating private variables. Also start to use nonatomic for properties, since some of the custom getters/setters were not implementing atomic locks anyways.

Update the logic for determining WWAN vs. WiFi vs. host not reachable.

Update project.pbxproj to allow building the tests from the command line.

Share a scheme for SPDY to allow setting the default test debugger from lldb to gdb.

Rename SessionKey -> SpdySessionKey. Create a separate file for SpdySessionKey. Add tests.

Commit:
Fix a couple bugs in SPDY.m

Also begin to migrate the code to a more modern style that uses properties/synthesize. Also start to use nonatomic for properties.

sorced-jim commented on SPDY/SPDY/SPDY.m:
```diff
@@ -158,6 +182,7 @@ - (SpdySession *)getSession:(NSURL *)url withError:(NSError **)error {
             return nil;
         }
         SPDY_LOG(@"Adding %@ to sessions (size = %u)", key, [sessions count] + 1);
+        currentStatus = [self.class reachabilityStatusForHost:key.host];
```
Why do you grab the currentStatus again?

erikchen replied:
it looks like currentStatus could change because of [session connect:url]

sorced-jim replied:
Humm, if that is the case we'd end up opening and closing a lot of connections, oh well.

erikchen replied:
nothing changed

sorced-jim commented on SPDY/SPDY/SPDY.m:
```diff
@@ -78,19 +79,19 @@ - (NSUInteger)hash {
 
 - (BOOL)isEqualToKey:(SessionKey *)other {
-    return [self.host isEqualToString:other.host] && (self.port == other.port || [self.port isEqualToNumber:other.port]);
+    return [self.host isEqualToString:other.host] && [self.port isEqualToNumber:other.port];
```
This is now wrong and you broke the tests.  Please run the tests before sending a pull request.

The self.port == other.port is needed for self and other both being nil.

erikchen replied:
fixed

sorced-jim commented on SPDY/SPDY/SPDY.m:
```diff
@@ -78,19 +79,19 @@ - (NSUInteger)hash {
     if (other == self)
         return YES;
-    if (!other || ![other isKindOfClass:[self class]])
+    if (!other || ![other isKindOfClass:[SessionKey class]])
```
Why?  self class should do the same thing.

erikchen replied:
changed back

sorced-jim commented on SPDY/SPDY/SPDY.m:
```diff
@@ -122,28 +123,51 @@ @implementation SPDY {
+}
+
++ (SpdyNetworkStatus)reachabilityStatusForHost:(NSString *)host {	
+    SpdyNetworkStatus status = kSpdyNotReachable;
```
The indentation looks weird here compared to the rest of lines.  I expect you'll want to reinvent this method.

erikchen replied:
I'm not seeing an indentation weirdness. Please describe in more detail?

sorced-jim commented on SPDY/SPDY/SPDY.m:
```diff
@@ -122,28 +123,51 @@ @implementation SPDY {
+    }
+    
+    // Host reachable by WWAN.
+    if ((flags & kSCNetworkReachabilityFlagsIsWWAN) == kSCNetworkReachabilityFlagsIsWWAN)
```
Why don't you do this check first and return kSpdyReachableViaWWAN?

erikchen replied:
done

sorced-jim commented:
Please update your commit description as well, it doesn't describe all that is being changed.

Commit:
Start responding to comments.

Set project setting Test after build = YES so that unit tests can be run from command line.

Commit:
Share the scheme for SPDY.

This allows us to change the test debugger from lldb to gdb, which is actually reasonable.

Commit:
Rename SessionKey -> SpdySessionKey. Create a separate file for SpdySessionKey. Add tests.

sorced-jim commented on SPDY/SPDY/SPDY.m:
```diff
@@ -122,28 +67,49 @@ @implementation SPDY {
+    // Host reachable. Connection is on-demand or on-traffic. No user intervention needed. Assume wifi.
+    if (((flags & kSCNetworkReachabilityFlagsConnectionOnDemand) != 0) ||
+        ((flags & kSCNetworkReachabilityFlagsConnectionOnTraffic) != 0)) {
+        if ((flags & kSCNetworkReachabilityFlagsInterventionRequired) == 0) {
```
It would be more consistent if you didn't have the braces for this if.

erikchen replied:
done

sorced-jim commented on SPDY/SPDY/SpdySessionKey.m:
```diff
@@ -0,0 +1,67 @@
+//      File: SpdySessionKey.m
+//
+//  Created by Erik Chen on 4/11/12.
+//  Copyright (c) 2012 Twist Inc. All rights reserved.
```
This is wrong, you need to switch to the apache license for this file.

erikchen replied:
okay. I had just copied SpdyInputStream

sorced-jim commented on SPDY/SPDY/SpdySessionKey.m:
```diff
@@ -0,0 +1,67 @@
+}
+
+@end
\ No newline at end of file
```
Please add a new line at the end of the file.

erikchen replied:
done

sorced-jim commented on SPDY/SPDYTests/SpdySessionKeyTests.h:
```diff
@@ -0,0 +1,13 @@
+//  Abstract: Tests for SpdySessionKey
+//
+//  Created by Erik Chen on 4/11/12.
+//  Copyright (c) 2012 Twist Inc. All rights reserved.
```
Same thing about the license.  I can fix this up if you want.

erikchen replied:
done

Commit:
Fix some style nits and add some missing licensing text.
